### ISSUE TITLE
Cannot write array literals

### ISSUE THREAD

**Issue by brabalan** [ocaml/tuareg#61]
If I try to evaluate
``` ocaml
let y = [| 1; 2 |] ;;
```
I get this
```
# let y = [| 1; 2 |;;
Characters 16-17:
  let y = [| 1; 2 |;;
                  ^
Error: Syntax error
```

For some reason the last `]` is not passed to the ocaml process, resulting in a syntax error.

**Comment by Chris00** [comment]
Do you use the latest version (available through opam)?  I cannot reproduce your problem:
```ocaml
# let y = [| 1; 2 |] ;;
val y : int array = [|1; 2|]
```

**Comment by brabalan** [comment]
I was using tuareg from melpa. I tried switching to the version from opam and I'm seeing the same thing. I can reproduce this with an emacs with no configuration (except for the loading of `tuareg-site-file`).

I'm trying to debug it and the problem is with `smie-forward-sexp` that stops too early. It returns `(t 19 "]")` but I'm not sure to understand what it means.

I am using emacs 24.5.1.

**Comment by Chris00** [comment]
Ah, I misunderstood your report.  You are trying to evaluate `let y = [| 1; 2 |]` with `C-x C-e` and the expression is not well detected.

**Comment by brabalan** [comment]
Yes, exactly.

**Comment by monnier** [comment]
I cannot reproduce this with the latest code from the Git repository.  I fixed a related issue some weeks (months?) ago, so maybe that was it.  Can someone still reproduce the problem with the latest code?

**Comment by Chris00** [comment]
He means having
```ocaml
let y = [| 1; 2 |]
```
in a buffer and trying to send it to the toplevel with `C-x C-e`.

**Comment by Chris00** [comment]
I just checked and it is fixed — likely by commit a551d07ab14d072aac752bd5f9003d1d54ed7490

**Comment by brabalan** [comment]
I confirm here as well. Thanks!
